Makefile.shared should pass SDKROOT to set-webkit-configuration

kmiller68 opened the pull request:
#### a6341cabd933197f63da3b9adcafed6dab5dd9e3
<pre>
Makefile.shared should pass SDKROOT to set-webkit-configuration
<a href="https://bugs.webkit.org/show_bug.cgi?id=273320">https://bugs.webkit.org/show_bug.cgi?id=273320</a>
<a href="https://rdar.apple.com/127114398">rdar://127114398</a>

Reviewed by David Kilzer.

This fixes the build scenario when Apple WebKittens try to do the OS build with
`make debug SDKROOT=macosx`. Previously we wouldn&apos;t forward the `SDKROOT` to
`set-webkit-configuration` which assumed the user wanted to use the .internal SDK.
This meant `set-webkit-configuration` would complain because it couldn&apos;t find
the Internal directory.

* Makefile.shared:

Canonical link: <a href="https://commits.webkit.org/278043@main">https://commits.webkit.org/278043@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/88a07e6856d66bb9c462cf5db625cf9fd7b8740d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49322 "Failed to checkout and rebase branch from PR 27801") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28604 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52353 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/52167 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45425 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34618 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26224 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/40292 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51423 "Failed to checkout and rebase branch from PR 27801") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/26151 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/42482 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21413 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/23608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7689 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/42624 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/45519 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/44166 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/54070 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/48811 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/24404 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/20584 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/47618 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/25676 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/42690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/46608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/10839 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26515 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/56306 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7078 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25399 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/11566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -914,7 +914,7 @@ sub determineXcodeSDKPlatformName {
         $xcodeSDKPlatformName = "maccatalyst";
     }
-    if (checkForArgumentAndRemoveFromARGVGettingValue("--sdk", \$sdk)) {
+    if (checkForArgumentAndRemoveFromARGVGettingValue("--sdk", \$sdk) || checkForArgumentAndRemoveFromARGVGettingValue("SDKROOT", \$sdk)) {
```
Not sure if we should support `set-webkit-configuration SDKROOT=...` but that's how folks pass it to `make` so maybe it's reasonable here too? Also happy to revert this.

ddkilzer replied:
No, `make` is different in how one passes arguments.

I don't think we need to support `set-webkit-configuration SDKROOT=blargh`.

kmiller68 commented on Makefile.shared:
```diff
@@ -28,6 +28,7 @@ ifeq (ON,$(ENABLE_LLVM_PROFILE_GENERATION))
 ifneq (,$(SDKROOT))
+	CONFIG_OPTIONS += --sdk=$(SDKROOT)
 	ifneq (,$(OVERRIDE_SDKROOT))
 		ifneq (default,$(OVERRIDE_SDKROOT))
```
Not sure what `OVERRIDE_SDKROOT` is used for. Maybe this is something I should be thinking about?

ddkilzer replied:
I would move the `CONFIG_OPTIONS` line to each place where the `--sdk=...` argument is added for `BUILD_WEBKIT_BASE` below instead of setting it here.  Also use `--sdk=$(OVERRIDE_SDK)` in the first case.

That way you're only adding `--sdk=...` to `CONFIG_OPTIONS` when you're also adding it to `BUILD_WEBKIT_BASE`, for consistency.

kmiller68 replied:
Seems reasonable. Done.

ddkilzer reviewed (changes requested):
Requested changes below for consistency when adding `--sdk=...` arguments.

ddkilzer reviewed (approved):
r=me

kmiller68 committed:
Makefile.shared should pass SDKROOT to set-webkit-configuration
https://bugs.webkit.org/show_bug.cgi?id=273320
rdar://127114398

Reviewed by David Kilzer.

This fixes the build scenario when Apple WebKittens try to do the OS build with
`make debug SDKROOT=macosx`. Previously we wouldn't forward the `SDKROOT` to
`set-webkit-configuration` which assumed the user wanted to use the .internal SDK.
This meant `set-webkit-configuration` would complain because it couldn't find
the Internal directory.

* Makefile.shared:

Canonical link: https://commits.webkit.org/278043@main
